### ISSUE TITLE
Avoid using spl_object_hash() for storing found posts

### ISSUE THREAD

**Issue by danielbachhuber** [10up/ElasticPress#1762]
In https://github.com/pantheon-systems/solr-power/pull/448, I discovered that `spl_object_hash()` [doesn't do what you'd expect it to do](https://www.php.net/manual/en/function.spl-object-hash.php#76220):

> Note that the contents (properties) of the object are NOT hashed by the function, merely its internal handle and handler table pointer. This is sufficient to guarantee that any two objects simultaneously co-residing in memory will have different hashes. Uniqueness is not guaranteed between objects that did not reside in memory simultaneously, for example:
>
> var_dump(spl_object_hash(new stdClass()), spl_object_hash(new stdClass()));
> 
> Running this alone will usually generate the same hashes, since PHP reuses the internal handle for the first stdClass after it has been dereferenced and destroyed when it creates the second stdClass.

tl;dr: You don't actually get a unique ID.

Given you need a unique ID for storing posts on a singleton in the test suite, I decided to instead generate the unique ID each `WP_Query` instance:

```php
/**
 * Generate a unique hash for the query at the beginning of the request.
 * We used to used spl_object_hash(), but it turned out not to be unique.
 *
 * @param object $query Existing query object.
 */
public function pre_get_posts( $query ) {
	if ( isset( $query->solr_query_id ) ) {
		return;
	}
	$query->solr_query_id = md5( mt_rand() );
}
``` 

You might want to update ElasticPress similarly.

**Comment by tlovett1** [comment]
Thanks @danielbachhuber. I'll take a look.
